### PR TITLE
[Merged by Bors] - improved p2p documentation

### PR DESCRIPTION
**Pull request by y0sher**
## Motivation

improve docs for better open source readiness

## Changes
added some documentation
updated documentation

### CODE REVIEW

File: p2p/message.go [hunk 1]
```diff
@@ -65,16 +66,19 @@ type ProtocolMessageMetadata struct {
...
 	Wrapped *service.DataMsgWrapper
 }
 
+// ProtocolMessage is a pair of metadata and a a payload.
 type ProtocolMessage struct {
 	Metadata *ProtocolMessageMetadata
 	Payload  *Payload
 }
 
+// CreatePayload is an helper function to format a payload for sending.
```
Comment (antonlerner) [inline p2p/message.go]: ```// CreatePayload is a helper function to format a payload for sending.```

Comment (antonlerner) [inline p2p/message.go]:
```suggestion
// CreatePayload is a helper function to format a payload for sending.
```

File: p2p/message.go [hunk 2]
```diff
@@ -91,6 +95,7 @@ func CreatePayload(data service.Data) (*Payload, error) {
 	return nil, fmt.Errorf("cant determine paylaod type")
 }
 
+// ExtractData is an helper function to extract the payload data from a message payload.
```
Comment (antonlerner) [inline p2p/message.go]:
```suggestion
// ExtractData is a helper function to extract the payload data from a message payload.
```

File: p2p/p2pcrypto/crypto.go [hunk 3]
```diff
@@ -86,6 +95,7 @@ func (k key) Open(encryptedMessage []byte) (out []byte, err error) {
 	return message, nil
 }
 
+// GenerateKeyPair generate private key and a public key derived from it.
```
Comment (antonlerner) [inline p2p/p2pcrypto/crypto.go]: maybe state what king of key type?

File: p2p/server/msgserver.go [hunk 4]
```diff
@@ -67,12 +71,14 @@ func NewMsgServer(network Service, name string, requestLifetime time.Duration, c
...
 }
 
+// Close stops the MessageServer
 func (p *MessageServer) Close() {
 	p.exit <- struct{}{}
 	<-p.exit
 	p.workerCount.Wait()
 }
 
+// readLoop readds incoming messages and matches them to requests or responses.
```
Comment (antonlerner) [inline p2p/server/msgserver.go]:
```suggestion
// readLoop reads incoming messages and matches them to requests or responses.
```

File: p2p/service/service.go [hunk 5]
```diff
@@ -59,30 +64,30 @@ type Service interface {
 	Shutdown()
 }
 
+// Data is a wrapper aroudn a message that can hold either raw bytes message or a req-res wrapper.
```
Comment (antonlerner) [inline p2p/service/service.go]:
```suggestion
// Data is a wrapper around a message that can hold either raw bytes message or a req-res wrapper.
```

File: p2p/swarm.go [hunk 6]
```diff
@@ -315,21 +328,20 @@ func (s *swarm) connectionPool() cPool {
 	return s.cPool
 }
 
+// SendWrappedMessage sends a which is wrapped in order to differentiate between request response and sub protocol messages.
```
Comment (antonlerner) [inline p2p/swarm.go]:
```suggestion
// SendWrappedMessage sends a wrapped message in order to differentiate between request response and sub protocol messages.
```

**Review by antonlerner — APPROVED** [review]

**Comment by y0sher** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by avive** [comment]
bors merge

**Pull request closed without merging**